hazelcast/src/main/java/com/hazelcast/internal/eventlog/Event.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.eventlog;
+
+import java.util.Map;
+
+/**
+ * Observable event in the system.
+ */
+public interface Event {
```

Comment:
If the API is supposed to be a general-purpose one, maybe adding an `origin` or `discriminator` would be useful. The `ILogger` implementation at https://github.com/hazelcast/hazelcast-enterprise/pull/3060/files#diff-c506b19dad6d70d9cbf3256a246b6660R20 has this information hard-wired, making all events to be treated as audit log events and log levels can be set only globally for all kind of events.